datafusion/expr/src/utils.rs

```diff
@@ -69,6 +69,94 @@ pub fn grouping_set_expr_count(group_expr: &[Expr]) -> Result<usize> {
     }
 }
 
+fn powerset<T>(slice: &[T]) -> Vec<Vec<&T>> {
```

Comment:
Can you please add a docstring explaining what the expected output of this function is intended to be?

For example, is it https://en.wikipedia.org/wiki/Power_set?

Reply:
> Can you please add a docstring explaining what the expected output of this function is intended to be?
> 
> For example, is it https://en.wikipedia.org/wiki/Power_set?

I will add comments to it

Reply:
There is a power set implementation in [itertools](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.powerset), did you consider using it?

Reply:
@yukkit  can you please answer @ozankabak 's question?